### DISCUSSION TITLE
Keybindings for toggle Project Panel

### DISCUSSION THREAD

**Discussion by oneforalone** [zed-industries/zed#6488] [Feature Requests]
Is there any keybindings for toggle the Project Panel, or is there a way to add a keymapping to Zed? It would be better if there's a complete cheatsheet for keybindings.

**Comment by JosephTLyons** [comment]
You should be able to customize the project panel using this keybinding name:

```json
workspace::ToggleLeftSidebar
```

The default binding is `cmd-b`

```json
{
  "context": "Workspace",
  "bindings": {
    "cmd-b": "workspace::ToggleLeftSidebar"
  }
},
```

You can view the default bindings by opening up the command palette (`cmd-shift-p`) and typing in `zed: open default keymap`.

**Reply by oneforalone** [reply]
Thanks, and apologies for the silly question cause i just check out the settings and find the defaults keybindings, also found that the customized keybinding config is `~/.config/zed/keymap.json`.

**Reply by nixpulvis** [reply]
Can I ask why the `"context"` needs to be "Workspace", while the command itself `"workspace::ToggleLeftSidebar"` contains that information?

**Reply by SomeoneToIgnore** [reply]
action qualifier and `context`  are quite orthogonal things: 

* `workspace::ToggleLeftSidebar` says that there's a `ToggleLeftSidebar` action that has been defined in the `workspace` namespace, code-wise, and anything that has a `workspace` code dependency might use it, dispatch it and define specific rules on "when" this dispatch happens.

* `context` in this context means "when" that action could be called — it could be `editor` and that would mean that only when some editor is focused this action (that got pulled from `workspace` crate) will be dispatched. When `workspace` is set there, it means "everywhere" as the workspace is a very fundamental thing that you always have focused.

**Reply by carukwbe** [reply]
seems no longer to work..
i get this error
![image](https://github.com/user-attachments/assets/f7351d7b-7c48-406c-987c-b90bb3212c74)

edit: 
It works, you just have to make sure to have no other conflicting keybindings. Error seems odd though.

**Reply by SomeoneToIgnore** [reply]
Seems that `workspace::ToggleLeftDock` is now the new name for this?
